### PR TITLE
Add VC Redist 2008 SP1 MFC to installer

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds the Visual C++ 2008 SP1 MFC Redistributable (vcredist) binary to the Salt installer. Required for older versions of Windows (2008, poss 2008r2, etc). The vcredist is required by Python 2.7. Python itself will not install unless that prereq is present. Since we're bundling python, we need to make sure that's on the system. This bundles the vcredist binary in the Salt installer for closed networks.

The Salt installer checks for the presence of vcredist and prompts the user to install VCRedist. If the user clicks `No`, salt will continue to install but will fail to start. The user will need to install the VCRedist manually.

If the Silent option is passed (`/S`) it just installs it.

### What issues does this PR fix or reference?
https://github.com/saltstack/zh/issues/1088
https://github.com/saltstack/salt/issues/37669#issuecomment-260673162

### Previous Behavior
Salt would fail to start if VCRedist is not found on the system.

### New Behavior
Salt installer prompts you to install VCRedist if it's not installed.

### Tests written?
NA

### CODE REVIEW

**Comment by damon-atkins** [comment]
> The Salt installer checks for the presence of vcredist and prompts the user to install VCRedist. If the user clicks No, salt will continue to install but will fail to start. The user will need to install the VCRedist manually.

Should just do it, if silent install

File: pkg/windows/installer/Salt-Setup.nsi [hunk 1]
```diff
@@ -367,6 +406,7 @@ Function .onInit
         Delete "$INSTDIR\uninst.exe"
         Delete "$INSTDIR\nssm.exe"
         Delete "$INSTDIR\salt*"
+        Delete "$INSTDIR\vcredist.ext"
```
Comment (lorengordon) [inline pkg/windows/installer/Salt-Setup.nsi]: `.ext` or `.exe`? also below on lines 520 and 561...

**Comment by lorengordon** [comment]
I am curious about this requirement. Didn't we make changes to the installer about a year ago that removed this package as a requirement? Keeping up with the security patches on this one was quite a pain.

**Comment by lorengordon** [comment]
Also, this will break for minions that do not have external Internet connectivity during the salt-minion install.

File: pkg/windows/installer/Salt-Setup.nsi (hunk 1 above)
Comment (twangboy) [inline pkg/windows/installer/Salt-Setup.nsi]: You're right. I'll update it.

**Comment by twangboy** [comment]
In previous iterations of including vcredist, it was downloaded. This iteration includes the most recent vcredist (Visual C++ 2008 SP1 Redistributable MFC) inside the package itself, no download. That's the most recent vcredist posted on 6/7/2011. <sarcasm>I'm sure it's totally secure and will never need to be patched.</sarcasm>

I've seen users reporting the need to install VCRedist over the months since we removed it. I was never able to replicate it until we got an msdn subscription and I was able to install a brand new Windows 2008 Server. So, we're adding it back, but putting the binary in the package.

**Comment by twangboy** [comment]
And a `/S` does install the VCRedist silently.

**Comment by lorengordon** [comment]
> I'm sure it's totally secure and will never need to be patched.

😂 

> This iteration includes the most recent vcredist (Visual C++ 2008 SP1 Redistributable MFC) inside the package itself, no download.

Oh, I see. This is in build_pkg.bat. That is what's new. When I was looking at similar logic in prior versions, vcredist was downloaded and installed during the salt-minion install (if it was missing). This looks fine. Thanks for the clarification.

https://github.com/saltstack/salt/pull/38726/files#diff-ebe1e59a1a7db9d50e760a7304bdc766R47

Also, please feel free to ping me if you need temporary access to a Windows Server at any point.

**Pull request merged**